packages/react-native/ReactCommon/react/nativemodule/webperformance/NativePerformanceObserver.cpp

```diff
@@ -25,74 +27,84 @@ NativePerformanceObserverModuleProvider(
 
 namespace facebook::react {
 
+std::optional<PerformanceEntryType> parseEntryType(int rawValue) {
+  if (rawValue < 0 || rawValue >= NUM_PERFORMANCE_ENTRY_TYPES) {
+    return std::nullopt;
+  }
+
+  return static_cast<PerformanceEntryType>(rawValue);
+}
+
 NativePerformanceObserver::NativePerformanceObserver(
     std::shared_ptr<CallInvoker> jsInvoker)
     : NativePerformanceObserverCxxSpec(std::move(jsInvoker)) {}
 
-void NativePerformanceObserver::startReporting(
-    jsi::Runtime& /*rt*/,
-    PerformanceEntryType entryType) {
-  auto reporter = PerformanceEntryReporter::getInstance();
+jsi::Object NativePerformanceObserver::createObserver(jsi::Runtime& rt, NativePerformanceObserverCallback callback) {
+  PerformanceObserverCallback cb = [callback = std::move(callback)](std::vector<PerformanceEntry>&& entries, size_t droppedEntriesCount) -> void {
+    callback.callWithPriority(SchedulerPriority::IdlePriority, std::move(entries), droppedEntriesCount);
```

Comment:
I don't think this signature matches how we're using the callback from native (where we're only passing the dropped entries count).